src/main/java/com/labsynch/labseer/service/StandardizationServiceImpl.java

```diff
@@ -207,70 +203,51 @@ private HashMap<String, Integer> saveDryRunStructures(HashMap<String, CmpdRegMol
 	}
 
 	@Override
-	@Transactional
 	public void populateStandardizationDryRunTable()
 			throws CmpdRegMolFormatException, IOException, StandardizerException {
-		List<Long> parentIds = Parent.getParentIds();
-
-		Long startTime = new Date().getTime();
-
-		int batchSize = propertiesUtilService.getStandardizationBatchSize();
-		int nonMatchingCmpds = 0;
-		int totalCount = parentIds.size();
-		logger.info("number of parents to check: " + totalCount);
-
-		Integer runNumber = StandardizationDryRunCompound.findMaxRunNumber().getSingleResult();
-		if (runNumber == null) {
-			runNumber = 1;
-		} else {
-			runNumber++;
-		}
-		float percent = 0;
-		int p = 0;
-		float previousPercent = percent;
-		previousPercent = percent;
-
-		// Split parent ids into groups of batchSize
-		List<List<Long>> parentIdGroups = SimpleUtil.splitArrayIntoGroups(parentIds, batchSize);
-
-		// Do a bulk standardization
-		for (List<Long> pIdGroup : parentIdGroups) {
-			dryrunStandardizeBatch(pIdGroup, nonMatchingCmpds, runNumber);
-			p = p + pIdGroup.size();
-
-			// Compute your percentage.
-			percent = (float) Math.floor(p * 100f / totalCount);
-			if (percent != previousPercent) {
-				Long currentTime = new Date().getTime();
-				// Output if different from the last time.
-				logger.info("populating standardization dry run table " + percent + "% complete (" + p + " of "
-						+ totalCount + ") average speed (rows/min):"
-						+ (p / ((currentTime - startTime) / 60.0 / 1000.0)));
-				currentTime = new Date().getTime();
-				logger.debug("Time Elapsed:" + (currentTime - startTime));
+		while (true) {
```

Comment:
Instead of selecting all Parent ids we now run a while true loop doing chunks of N parent ids until there are non left to process and all ids are in the dry run standardization table,  if there are 0 returned, then it needs to do a final check to make sure all ids are processed, or wait a few seconds and try again.  


Besides freeing up memory so that not even parent id has to be held in memory, this also allows us to continue the process if the processing fails for some reason.  If that happens a new roo worker will come up, check if standardization dry run is "running" and will start populating the dry run table until all ids are present.